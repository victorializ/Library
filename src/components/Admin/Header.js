import React from 'react';
import { Link } from 'react-router-dom';
import { Menu, Header } from 'semantic-ui-react';

import { constants } from '../../i18n';

const AdminHeader = () => 
    <Menu widths={2}>
        <Menu.Item as={Link} to='/admin/order'>
            <Header>
                {constants.order}
            </Header>
        </Menu.Item>
        <Menu.Item as={Link} to='/admin/manage'>
            <Header>
                {constants.manage}
            </Header>
        </Menu.Item>
    </Menu>  

export { AdminHeader };