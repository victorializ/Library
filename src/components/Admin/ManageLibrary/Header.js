import React from 'react';
import { Link } from 'react-router-dom';
import { Menu, Header } from 'semantic-ui-react';

import { constants } from '../../../i18n';

function ManageHeader() {
    return (
      <Menu widths={3}>
        <Menu.Item as={Link} to='/admin/manage/books'>
          <Header>
            {constants.books}
          </Header>
        </Menu.Item>
        <Menu.Item as={Link} to='/admin/manage/authors'>
          <Header>
            {constants.bookAuthors}
          </Header>
        </Menu.Item>
        <Menu.Item as={Link} to='/admin/manage/users'>
          <Header>
            {constants.userslist}
          </Header>
        </Menu.Item>
      </Menu>
    )
}

export { ManageHeader };