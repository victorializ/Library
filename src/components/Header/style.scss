@import '../../assets/variables.scss';

header {
    .ui.menu {
        margin: 0;
        height: 10vh;
    }
    .item {
        border: 0;
        width: 10vw;
        @include centered-horizontal-container;
    }
}

.app__header {
    &__logo {
        margin-right: auto;
    }
}