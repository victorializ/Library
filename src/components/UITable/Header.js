import React from 'react';
import { Link } from 'react-router-dom';
import { Menu, Header as SHeader } from 'semantic-ui-react'

import { CurrentUser } from '../User/User';
import { constants } from '../../i18n';

function Header() {
  return (
    <header>
        <Menu className='app__header'>
            <Menu.Item as={Link} to='/' className='app__header__logo'>
                <SHeader>
                    {constants.title}
                </SHeader>
            </Menu.Item>
            <Menu.Item className='app__header__current-user'>
                <CurrentUser />
            </Menu.Item> 
            <Menu.Item as={Link} to='/login'>
                <SHeader>
                    {constants.login}
                </SHeader>
            </Menu.Item>
            <Menu.Item as={Link} to='/author'>
                <SHeader>
                    {constants.bookAuthors}
                </SHeader>
            </Menu.Item>
            <Menu.Item as={Link} to='/book'>
                <SHeader>
                    {constants.books}
                </SHeader>
            </Menu.Item>
        </Menu>
    </header>
  );
}

export {  Header };