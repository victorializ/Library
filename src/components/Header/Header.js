import React from 'react';

import { Menu, Header as SHeader } from 'semantic-ui-react'
import { Link } from 'react-router-dom';

import { constants  } from '../../i18n';
import { UserIcon } from '../../components';

import './style.scss';

function Header() {
  return (
    <header>
        <Menu>
            <Menu.Item as={Link} to='/' className='app__header__logo'>
                <SHeader>
                    {constants.title}
                </SHeader>
            </Menu.Item>
            <Menu.Item>
                <SHeader>
                    <UserIcon />
                </SHeader>
            </Menu.Item> 
            <Menu.Item as={Link} to='/login'>
                <SHeader>
                    {constants.login}
                </SHeader>
            </Menu.Item>
            <Menu.Item as={Link} to='/author'>
                <SHeader>
                    {constants.bookAuthors}
                </SHeader>
            </Menu.Item>
            <Menu.Item as={Link} to='/book'>
                <SHeader>
                    {constants.books}
                </SHeader>
            </Menu.Item>
        </Menu>
    </header>
  );
}

export {  Header };