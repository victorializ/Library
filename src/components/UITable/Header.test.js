import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header component', () => {
    it('renders correctly', () => {
        expect(shallow(<Header />)).toMatchSnapshot();
    });   
});