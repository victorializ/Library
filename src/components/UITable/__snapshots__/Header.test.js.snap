// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header>
  <Menu
    className="app__header"
  >
    <MenuItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      className="app__header__logo"
      to="/"
    >
      <Header>
        Library
      </Header>
    </MenuItem>
    <MenuItem
      className="app__header__current-user"
    >
      <Connect(User) />
    </MenuItem>
    <MenuItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/login"
    >
      <Header>
        Login
      </Header>
    </MenuItem>
    <MenuItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/author"
    >
      <Header>
        Authors
      </Header>
    </MenuItem>
    <MenuItem
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/book"
    >
      <Header>
        Books
      </Header>
    </MenuItem>
  </Menu>
</header>
`;
